edn/src/types.rs

```diff
@@ -321,44 +377,45 @@ mod test {
     use std::iter::FromIterator;
     use std::f64;
 
-    use symbols;
-    use parse;
     use num::BigInt;
     use ordered_float::OrderedFloat;
 
+    use parse;
+
     #[test]
     fn test_value_from() {
-        assert_eq!(Value::from(42f64), Value::Float(OrderedFloat::from(42f64)));
+        assert_eq!(Value::from_float(42f64), Value::Float(OrderedFloat::from(42f64)));
+        assert_eq!(Value::from_ordered_float(OrderedFloat::from(42f64)), Value::Float(OrderedFloat::from(42f64)));
         assert_eq!(Value::from_bigint("42").unwrap(), Value::BigInteger(BigInt::from(42)));
     }
 
     #[test]
     fn test_print_edn() {
         let string = "[ 1 2 ( 3.14 ) #{ 4N } { foo/bar 42 :baz/boz 43 } [ ] :five :six/seven eight nine/ten true false nil #f NaN #f -Infinity #f +Infinity ]";
```

Comment:
If I duplicate the hierarchy as described above, then yes, mapping ValueAndSpan onto Value would be possible here. Otherwise not (because Value::Collection was over ValueAndSpan)